src/main/java/Duke.java

```diff
@@ -1,10 +1,122 @@
+import java.util.Scanner;
+
 public class Duke {
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public String getStatusIcon() {
+            return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+        }
+        public void markAsDone(){
+            this.isDone = true;
+        }
+        public String toString() {
+            return description;
+        }
+    }
+
+    public static class Todo extends Task{
+        public Todo(String description){
+            super(description);
+        }
+        @Override
+        public String toString() {
+            return "[T][" + getStatusIcon() + "] " + super.toString();
+        }
+    }
+    public static class Deadline extends Task{
+        protected String by;
+        public Deadline(String description, String by){
+            super(description);
+            this.by = by;
+        }
+        @Override
+        public String toString() {
+            return "[D][" + getStatusIcon() + "] " + super.toString() + "(by:" + by + ")";
+        }
+    }
+
+    public static class Event extends Task{
+        protected String at;
+        public Event(String description, String at){
+            super(description);
+            this.at = at;
+        }
+        @Override
+        public String toString() {
+            return "[E][" + getStatusIcon() + "] " + super.toString() + "(at:" + at + ")";
+        }
+    }
+
     public static void main(String[] args) {
+        Scanner in = new Scanner(System.in);
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?\n");
+        Task[] items = new Task[100];
+        int itemsCount = 0;
+        String input = in.nextLine();
+        while (true) {
+            if (input.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            }
+            else if (input.equals("list")){
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 0; i<itemsCount; i++){
+//                    System.out.println(i+1 + ".["+ items[i].getStatusIcon() +"] "+ items[i].description);
+                    System.out.println(String.valueOf(i+1) + ". " + items[i]);
+                }
+                input = in.nextLine();
+            }
+            else if (input.contains("done")){
```

Comment:
Consider refactoring to use more constants and reduce the number of magic literals.